### PR TITLE
Add Plugin Metadata

### PR DESCRIPTION
**Pull request by helenkzhang**
Creates the plugin manager project and defines the metadata of a local plugin

Example serialized file looks like below 
(Information is manually entered)

```
{
  "Id": "LinuxAndroidTools",
  "Version": "1.0.0",
  "DisplayName": "Linux Tool",
  "Description": "Linux Tool Bundles",
  "Owners": [
    {
      "Name": "Someone",
      "Address": null,
      "EmailAddresses": [
        "abc@outlook.com",
        "xyz@outlook.com"
      ],
      "PhoneNumbers": [
        "123-333-4566"
      ]
    }
  ],
  "SdkVersion": "1.0.0",
  "ProcessingSources": [
    {
      "Guid": "98608154-6231-4f25-903a-5e440574ab45",
      "Name": "LTTng",
      "Version": "2.0.0.0",
      "Description": null,
      "AboutInfo": {
        "Owners": [
          {
            "Name": "ps1",
            "Address": null,
            "EmailAddresses": [
              "processingsource1@micrsoft.com"
            ],
            "PhoneNumbers": [
              "123-456-7890"
            ]
          }
        ],
        "ProjectInfo": {
          "Uri": "github.com"
        },
        "LicenseInfo": {
          "Name": "MIT",
          "Uri": "license.com",
          "Text": "MIT"
        },
        "CopyrightNotice": "Copyright notice",
        "AdditionalInformation": null
      },
      "AvailableTables": [
        {
          "Guid": "f8f953b2-dfed-448b-87dd-a15cb6dd1d6d",
          "Name": "Trace Stats",
          "Description": "Trace Stats",
          "Category": "Unkown",
          "IsMetadataTable": false
        }
      ],
      "SupportedDataSources": [
        {
          "Name": ".ctf",
          "Description": "Processes LTTng CTF data"
        }
      ]
    },
    {
      "Guid": "a9ac39bc-2d07-4a01-b9b5-13a02611f5f2",
      "Name": "WaLinuxAgent",
      "Version": "1.0.0.0",
      "Description": null,
      "AboutInfo": null,
      "AvailableTables": null,
      "SupportedDataSources": [
        {
          "Name": ".log",
          "Description": "Linux WaLinuxAgent Cloud Provisioning Log"
        }
      ]
    }
  ],
  "DataCookers": [
    {
      "Name": "dc1",
      "Description": "dc1"
    }
  ],
  "ExtensibleTables": [
    {
      "Guid": "6e8ad993-cef3-40c2-826a-078e21694c8f",
      "Name": "Extensible Table",
      "Description": "extensible table",
      "Category": "Unkown",
      "IsMetadataTable": false
    }
  ]
}
```

### CODE REVIEW

File: src/PluginManager/Microsoft.Performance.Toolkit.PluginManager.Core/Packaging/Metadata/PluginMetadata.cs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+{
+    /// <summary>
+    /// Represents the metadata of a plugin
+    /// </summary>
+    public sealed class PluginMetadata
+    {
+        /// <summary>
+        /// The group this plugin belongs to
+        /// </summary>
+        public string Group { get; set; }
```
Comment (mslukebo) [inline src/PluginManager/Microsoft.Performance.Toolkit.PluginManager.Core/Packaging/Metadata/PluginMetadata.cs:17]: Can you elaborate on this? I'm not sure what Group means here

File: src/PluginManager/Microsoft.Performance.Toolkit.PluginManager.Core/Packaging/Metadata/PluginMetadata.cs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+        /// <summary>
+        /// A user friendly description of this plugin
+        /// </summary>
+        public string Description { get; set; }
+
+        /// <summary>
+        /// The authors of this plugin
+        /// </summary>
+        public IEnumerable<string> Authors { get; set; }
```
Comment (mslukebo) [inline src/PluginManager/Microsoft.Performance.Toolkit.PluginManager.Core/Packaging/Metadata/PluginMetadata.cs:42]: I'm wondering if these should be some `PluginAuthor` type that has stuff like contact info on it or something. What do you think?

File: src/PluginManager/Microsoft.Performance.Toolkit.PluginManager.Core/Microsoft.Performance.Toolkit.PluginManager.Core.csproj [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+  <ItemGroup>
+    <Folder Include="Discovery\" />
+    <Folder Include="Manager\" />
+    <Folder Include="Registry\" />
+    <Folder Include="Publishing\" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <EditorConfigFiles Remove="D:\playground\microsoft-performance-toolkit-sdk\src\PluginManager\Microsoft.Performance.Toolkit.PluginManager.Core\.editorconfig" />
```
Comment (mslukebo) [inline src/PluginManager/Microsoft.Performance.Toolkit.PluginManager.Core/Microsoft.Performance.Toolkit.PluginManager.Core.csproj:15]: Did you mean to leave this and the one below in?

File: src/PluginManager/Microsoft.Performance.Toolkit.PluginManager.Core/.editorconfig [hunk 4]
```diff
@@ -0,0 +1,148 @@
+[*.cs]
```
Comment (mslukebo) [inline src/PluginManager/Microsoft.Performance.Toolkit.PluginManager.Core/.editorconfig:1]: We should just have a single .editorconfig for the whole sol'n. I think we should make a separate PR into main that adds one

File: src/PluginManager/Microsoft.Performance.Toolkit.PluginManager.Core/Packaging/Metadata/Architecture.cs [hunk 5]
```diff
@@ -0,0 +1,14 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.Performance.Toolkit.PluginManager.Core
+{
+    public enum Architecture
```
Comment (mslukebo) [inline src/PluginManager/Microsoft.Performance.Toolkit.PluginManager.Core/Packaging/Metadata/Architecture.cs:6]: We should just require plugins to be AnyCPU and get rid of this :)

File: src/PluginManager/Microsoft.Performance.Toolkit.PluginManager.Core/Packaging/Metadata/TableMetadata.cs [hunk 6]
```diff
@@ -0,0 +1,26 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.Performance.Toolkit.PluginManager.Core.Packaging.Metadata
+{
+    /// <summary>
+    /// Represents the metadata of a table
+    /// </summary>
+    public sealed class TableMetadata
```
Comment (mslukebo) [inline src/PluginManager/Microsoft.Performance.Toolkit.PluginManager.Core/Packaging/Metadata/TableMetadata.cs:9]: Do we want an `IsMetadataTable` property?

File: src/PluginManager/Microsoft.Performance.Toolkit.PluginManager.Core/Packaging/Metadata/DataSourceType.cs [hunk 7]
```diff
@@ -0,0 +1,12 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.Performance.Toolkit.PluginManager.Core.Packaging.Metadata
+{
+    public enum DataSourceType
+    {
+        FileDataSource,
+        ExtensionlessFileDataSource,
```
Comment (mslukebo) [inline src/PluginManager/Microsoft.Performance.Toolkit.PluginManager.Core/Packaging/Metadata/DataSourceType.cs:9]: I'm wondering if instead of having a bunch of information about the actual supported data sources we just want something on `PluginMetadata` like

`PluginMetadata.SupportsFolders`
`PluginMetadata.SupportedExtensions`
`PluginMetadata.SupportsExtensionlessFiles`

From the user's perspective, they really only care about what can be processed by installing the plugin. Thoughts?

Comment (mslukebo) [reply src/PluginManager/Microsoft.Performance.Toolkit.PluginManager.Core/Packaging/Metadata/DataSourceType.cs:9]: Or putting those properties on `ProcessingSourceMetadata` instead of `PluginMetadat`

File: src/PluginManager/Microsoft.Performance.Toolkit.PluginManager.Core/Microsoft.Performance.Toolkit.PluginManager.Core.csproj (hunk 3 above)
Comment (helenkzhang) [reply src/PluginManager/Microsoft.Performance.Toolkit.PluginManager.Core/Microsoft.Performance.Toolkit.PluginManager.Core.csproj:15]: This got auto populated when I added an .editorconfig file to a project (right click add new EditorConfig). Not sure why it added a `Remove` here. I removed this anyways. Makes sense to add it to the sln in another PR.

File: src/PluginManager/Microsoft.Performance.Toolkit.PluginManager.Core/Packaging/Metadata/DataSourceType.cs (hunk 7 above)
Comment (helenkzhang) [reply src/PluginManager/Microsoft.Performance.Toolkit.PluginManager.Core/Packaging/Metadata/DataSourceType.cs:9]: Let's chat about it in call.

File: src/PluginManager/Microsoft.Performance.Toolkit.PluginManager.Core/Packaging/Metadata/PluginMetadata.cs (hunk 1 above)
Comment (helenkzhang) [reply src/PluginManager/Microsoft.Performance.Toolkit.PluginManager.Core/Packaging/Metadata/PluginMetadata.cs:17]: Removed for now

File: src/PluginManager/Microsoft.Performance.Toolkit.PluginManager.Core/Packaging/Metadata/PluginMetadata.cs (hunk 2 above)
Comment (helenkzhang) [reply src/PluginManager/Microsoft.Performance.Toolkit.PluginManager.Core/Packaging/Metadata/PluginMetadata.cs]: Added

File: src/PluginManager/Microsoft.Performance.Toolkit.PluginManager.Core/Packaging/Metadata/DataSourceType.cs (hunk 7 above)
Comment (helenkzhang) [reply src/PluginManager/Microsoft.Performance.Toolkit.PluginManager.Core/Packaging/Metadata/DataSourceType.cs]: Since `description` would probably be needed in the UI, I'm going to keep it as is for now.
